### DISCUSSION TITLE
Can anyone please point me to a few good examples of `SwarmBuilder::with_tokio_executor()`?

### DISCUSSION THREAD

**Discussion by harsh4786** [libp2p/rust-libp2p#4528] [Q&A]
I tried searching for some example implementations of  swarm using a tokio executor and I want to specifically refer to how events are implemented in those examples.

**Comment by thomaseizinger** [comment]
Does this help? https://github.com/libp2p/rust-libp2p/blob/master/misc/server/src/main.rs

Also, on a related note, we'd actually like to move to prefering tokio everywhere: https://github.com/libp2p/rust-libp2p/issues/4449

**Comment by thomaseizinger** [comment]
> and I want to specifically refer to how events are implemented in those examples.

How events are handled is agnostic to which executor you are using. Picking the right executor only matters because tokio for example requires all its IO futures to run on a runtime thread because only those have a reactor running.
